### PR TITLE
AccessList in JSONRPC. And enabling github action tests

### PR DESCRIPTION
**Pull request by rakita**
When removing windows build we removed test execution, reverting this change.

### CODE REVIEW

**Review by adria0 — CHANGES_REQUESTED** [review]
As far as I see this is going to change some JSON-RPC responses? This is a breaking change for clients, and JSON codification for typed tx ids and access lists will be defined sooner in another EIP.

**Comment by rakita** [comment]
Other clients already enabled eip2718 in their RPC. This is example from Geth: https://github.com/ethereum/go-ethereum/pull/21502#issuecomment-758659399 
But I will need to change how we present accessList address/key in json.

**Comment by rakita** [comment]
and i will see to update our wiki

File: crates/rpc/src/v1/types/transaction_request.rs [hunk 1]
```diff
@@ -52,7 +52,8 @@ pub struct TransactionRequest {
     /// Delay until this block condition.
     pub condition: Option<TransactionCondition>,
     /// Access list
-    pub access_list: Option<AccessList>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub access_list: Option<Vec<AccessListItem>>,
```
Comment (adria0) [inline crates/rpc/src/v1/types/transaction_request.rs:56]:
```suggestion
    pub access_list: Option<AccessList>,
```

File: crates/rpc/src/v1/types/transaction.rs [hunk 2]
```diff
@@ -74,7 +73,8 @@ pub struct Transaction {
     /// Transaction activates at specified block.
     pub condition: Option<TransactionCondition>,
     /// optional access list
-    pub access_list: AccessList,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub access_list: Option<Vec<AccessListItem>>,
```
Comment (adria0) [inline crates/rpc/src/v1/types/transaction.rs:77]:
```suggestion
    pub access_list: Option<AccessList>,
```

File: crates/rpc/src/v1/types/call_request.rs [hunk 3]
```diff
@@ -42,7 +45,8 @@ pub struct CallRequest {
     /// Nonce
     pub nonce: Option<U256>,
     /// Access list
-    pub access_list: Option<AccessList>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub access_list: Option<Vec<AccessListItem>>,
```
Comment (adria0) [inline crates/rpc/src/v1/types/call_request.rs:49]:
```suggestion
    pub access_list: Option<AccessList>,
```

File: crates/rpc/src/v1/helpers/requests.rs [hunk 4]
```diff
@@ -109,7 +109,7 @@ pub struct CallRequest {
     /// Nonce
     pub nonce: Option<U256>,
     /// Access list
-    pub access_list: Option<AccessList>,
+    pub access_list: Option<Vec<AccessListItem>>,
```
Comment (adria0) [inline crates/rpc/src/v1/helpers/requests.rs:112]:
```suggestion
    pub access_list: Option<AccessList>,
```

**Review by adria0 — APPROVED** [review]

**Review by karim-agha — APPROVED** [review]

**Pull request merged**